odev/common/connectors/rpc.py

```diff
@@ -241,18 +253,36 @@ def __repr__(self) -> str:
     @property
     def url(self) -> str:
         """Return the URL to the external service."""
+        if not self.database.url:
+            raise ConnectorError(
+                f"URL not set for {self.database.platform.display} database {self.database.name!r}", self
+            )
+
         return urlparse(self.database.url, scheme="https").netloc.partition(":")[0]
 
     @property
     def port(self) -> int:
         """Return the port to the external service."""
+        if not self.database.rpc_port:
```

Comment:
Why don't we set use a default value here instead ?  (it's almost always 443 except locally)

Reply:
It's already set [here](https://github.com/odoo-odev/odev/blob/main/odev/common/databases/remote.py#L129-L130) for remote databases, but for local databases it is only set if the database is running (because it depends on the process if using `-p`). I think raising here is more helping than getting the generic authentication error from odoolib